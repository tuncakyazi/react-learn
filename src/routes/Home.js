import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => (
  <div>
    <h2>Home</h2>
    <ul>
      <li>
        <Link to="/product">Product</Link>
      </li>
    </ul>
  </div>
);

export default Home;
