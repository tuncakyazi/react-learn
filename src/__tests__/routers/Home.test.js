import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Home from './../../routes/Home';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const snap = shallow(<Home />);

  expect(snap).toMatchSnapshot();
});
