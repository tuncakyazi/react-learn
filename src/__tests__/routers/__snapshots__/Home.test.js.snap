// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Home
</h2>,
        <ul>
          <li>
                    <Link
                              replace={false}
                              to="/product"
                    >
                              Product
                    </Link>
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Home",
        },
        "ref": null,
        "rendered": "Home",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li>
            <Link
                        replace={false}
                        to="/product"
            >
                        Product
            </Link>
</li>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/product"
>
              Product
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Product",
              "replace": false,
              "to": "/product",
            },
            "ref": null,
            "rendered": "Product",
            "type": [Function],
          },
          "type": "li",
        },
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Home
</h2>,
          <ul>
            <li>
                        <Link
                                    replace={false}
                                    to="/product"
                        >
                                    Product
                        </Link>
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
          },
          "ref": null,
          "rendered": "Home",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              <Link
                            replace={false}
                            to="/product"
              >
                            Product
              </Link>
</li>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/product"
>
                Product
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Product",
                "replace": false,
                "to": "/product",
              },
              "ref": null,
              "rendered": "Product",
              "type": [Function],
            },
            "type": "li",
          },
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
